src/test/java/com/google/error_analyzer/PaginationServletTest.java

```diff
@@ -0,0 +1,141 @@
+    https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.*/
+package com.google.error_analyzer;
+
+import com.google.common.collect.ImmutableList;
+import com.google.error_analyzer.backend.LogDao;
+import com.google.error_analyzer.backend.LogDaoHelper;
+import com.google.error_analyzer.backend.PaginationServlet;
+import com.google.error_analyzer.data.SearchErrors;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.*;
+import javax.servlet.http.HttpServletResponse;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.SearchHits;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.runners.JUnit4;
+import org.junit.runner.RunWith;
+import org.junit.Test;
+import org.mockito.ArgumentMatchers;
+import org.mockito.InjectMocks;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.mockito.Mock;
+import org.mockito.Mockito.*;
+import org.mockito.MockitoAnnotations;
+import org.mockito.quality.Strictness;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.springframework.test.util.ReflectionTestUtils;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.BDDMockito.any;
+
+
+@RunWith(MockitoJUnitRunner.class)
+public final class PaginationServletTest {
+
+    private String fileName = "file";
+    private String fileType1 = "errors";
+    private String filetType2 = "logs";
+    private int page1 = 1;
+    private int page2 = 2;
+
+    @Mock
+    LogDao logDao;
+
+    @Mock
+    LogDaoHelper logDaoHelper;
+
+    @InjectMocks
+    PaginationServlet pagination;
+
+    @Rule
+    public MockitoRule rule = MockitoJUnit.rule();
+
+    @Before
+    public void setUp() {
+        pagination = new PaginationServlet();
+        ReflectionTestUtils.setField(pagination, "logDao", logDao);
+        ReflectionTestUtils.setField(pagination, "logDaoHelper", 
+        logDaoHelper);
+        MockitoAnnotations.initMocks(this);
+    }
+
+    // database related mocked functions
+    private void databaseHelper() throws IOException {
+        ImmutableList<String> immutableListId = ImmutableList.of("2");
+        ImmutableList<String> immutableListContent = 
+            ImmutableList.of("error2");
+        when(logDaoHelper.hitId(any())).thenReturn(immutableListId);
+        when(logDaoHelper.hitFieldContent(any(),any())).
+            thenReturn(immutableListContent);
+    }
+
+    // access private method addFetchResultToData
+    private Object getPrivateAddErrorFixesAndHighlights(
```

Comment:
Generally, we don't test private functions.
Testing the public functions should be sufficient (as internally they end up calling the private functions).

Testing doPost should be sufficient.